File: Javascript/1)Basic Logic/1.4 Type conversion.js
// Number
let num = 25;
console.log("Number:", num);
console.log("Number to String:", String(num));
console.log("Number to Boolean:", Boolean(num));

// String
let str = "100";
console.log("\nString:", str);
console.log("String to Number:", Number(str));
console.log("String to Integer:", parseInt(str));
console.log("String to Float:", parseFloat("100.50"));
console.log("String to Boolean:", Boolean(str));

// Boolean
let flag = true;
console.log("\nBoolean:", flag);
console.log("Boolean to Number:", Number(flag));
console.log("Boolean to String:", String(flag));

// Float
let decimal = 12.75;
console.log("\nFloat:", decimal);
console.log("Float to Integer:", parseInt(decimal));
console.log("Float to String:", String(decimal));

// Array
let arr = [10, 20, 30];
console.log("\nArray:", arr);
console.log("Array to String:", String(arr));
console.log("Array using join():", arr.join("-"));

// Object
let student = {
    name: "Sona",
    age: 20
};
console.log("\nObject:", student);
console.log("Object to String:", JSON.stringify(student));

// Character (ASCII / Unicode)
let ch = "A";
console.log("\nCharacter:", ch);
console.log("ASCII/Unicode:", ch.charCodeAt(0));

let code = 66;
console.log("Character from Code:", String.fromCharCode(code));

// Date to String
let today = new Date();
console.log("\nDate:", today);
console.log("Date to String:", today.toString());

// Number to Binary, Octal and Hexadecimal
let n = 25;
console.log("\nNumber:", n);
console.log("Binary:", n.toString(2));
console.log("Octal:", n.toString(8));
console.log("Hexadecimal:", n.toString(16));

File: Python/1)Basic Logic/1.4 Type conversion.py

# Integer
num = 25
print("Integer:", num)
print("int -> float:", float(num))
print("int -> str:", str(num))
print("int -> bool:", bool(num))
print()

# Float
decimal = 15.75
print("Float:", decimal)
print("float -> int:", int(decimal))
print("float -> str:", str(decimal))
print("float -> bool:", bool(decimal))
print()

# String
text = "100"
print("String:", text)
print("str -> int:", int(text))
print("str -> float:", float(text))
print("str -> bool:", bool(text))
print()

# Boolean
flag = True
print("Boolean:", flag)
print("bool -> int:", int(flag))
print("bool -> float:", float(flag))
print("bool -> str:", str(flag))
print()

# List
numbers = [10, 20, 30]
print("List:", numbers)
print("list -> tuple:", tuple(numbers))
print("list -> set:", set(numbers))
print("list -> str:", str(numbers))
print()

# Tuple
t = (1, 2, 3)
print("Tuple:", t)
print("tuple -> list:", list(t))
print("tuple -> set:", set(t))
print("tuple -> str:", str(t))
print()

# Set
s = {5, 6, 7}
print("Set:", s)
print("set -> list:", list(s))
print("set -> tuple:", tuple(s))
print("set -> str:", str(s))
print()

# Dictionary
student = {"name": "Priyanka", "age": 21}
print("Dictionary:", student)
print("dict -> list:", list(student))          # Keys only
print("dict -> tuple:", tuple(student))        # Keys only
print("dict -> set:", set(student))            # Keys only
print("dict -> str:", str(student))
print()

# Character conversions using ASCII
ch = 'A'
print("Character:", ch)
print("char -> ASCII:", ord(ch))

ascii_value = 66
print("ASCII -> char:", chr(ascii_value))
print()

# Binary, Octal and Hexadecimal
n = 25
print("Number:", n)
print("Binary:", bin(n))
print("Octal:", oct(n))
print("Hexadecimal:", hex(n))
print()

# String to List, Tuple and Set
word = "Python"
print("String:", word)
print("str -> list:", list(word))
print("str -> tuple:", tuple(word))
print("str -> set:", set(word))